source/jormungandr/jormungandr/interfaces/v1/serializer/geocode_json.py

```diff
@@ -216,12 +216,19 @@ class StopAreaSerializer(serpy.DictSerializer):
     timezone = NestedPropertyField(attr='properties.geocoding.timezone')
     commercial_modes = jsonschema.MethodField()
     physical_modes = jsonschema.MethodField()
+    codes = jsonschema.MethodField()
+    properties = jsonschema.MethodField()
```

Comment:
you can create three class new classes for code, mode and properties

```python

class NestedDictGenericField(DictGenericSerializer, NestedPropertyField):
    pass


class NestedDictCodeField(DictCodeSerializer, NestedPropertyField):
    pass


class NestedPropertiesField(NestedPropertyField):
    def to_value(self, value):
        return {p.get('key'): p.get('value') for p in value}
```

then use them for *_modes, codes and properties:

```python

commercial_modes = NestedDictGenericField(attr='properties.geocoding.commercial_modes', many=True)
physical_modes = NestedDictGenericField(attr='properties.geocoding.physical_modes', many=True)
codes = NestedDictCodeField(attr='properties.geocoding.codes', many=True)
properties = NestedPropertiesField(attr='properties.geocoding.properties')

```